addons/survey/wizard/survey_email_compose_message.py

```diff
@@ -1,70 +1,62 @@
+    date_deadline = fields.Date(string="Deadline to which the invitation to respond is valid", help="Deadline to which the invitation to respond for this survey is valid. If the field is empty, the invitation is still valid.")
+
+    @api.multi
+    def _compute_public_url(self):
+        for wizard in self:
+            wizard.public_url = wizard.survey_id.public_url
+
+    @api.multi
+    def _compute_public_url_html(self):
         """ Compute if the message is unread by the current user """
-        urls = self._get_public_url(cr, uid, ids, name, arg, context=context)
-        for key, url in urls.items():
-            urls[key] = '<a href="%s">%s</a>' % (url, _("Click here to start survey"))
-        return urls
-
-    _columns = {
-        'survey_id': fields.many2one('survey.survey', 'Survey', required=True),
-        'public': fields.selection([('public_link', 'Share the public web link to your audience.'),
-                                    ('email_public_link', 'Send by email the public web link to your audience.'),
-                                    ('email_private', 'Send private invitation to your audience (only one response per recipient and per invitation).')],
-            string='Share options', required=True),
-        'public_url': fields.function(_get_public_url, string="Public url", type="char"),
-        'public_url_html': fields.function(_get_public_url_html, string="Public HTML web link", type="char"),
-        'partner_ids': fields.many2many('res.partner',
-            'survey_mail_compose_message_res_partner_rel',
-            'wizard_id', 'partner_id', 'Existing contacts'),
-        'attachment_ids': fields.many2many('ir.attachment',
-            'survey_mail_compose_message_ir_attachments_rel',
-            'wizard_id', 'attachment_id', 'Attachments'),
-        'multi_email': fields.text(string='List of emails', help="This list of emails of recipients will not converted in contacts. Emails separated by commas, semicolons or newline."),
-        'date_deadline': fields.date(string="Deadline to which the invitation to respond is valid", help="Deadline to which the invitation to respond for this survey is valid. If the field is empty, the invitation is still valid."),
-    }
-
-    _defaults = {
-        'public': 'public_link',
-        'survey_id': lambda self, cr, uid, ctx={}: ctx.get('model') == 'survey.survey' and ctx.get('res_id') or None
-    }
-
-    def default_get(self, cr, uid, fields, context=None):
-        res = super(survey_mail_compose_message, self).default_get(cr, uid, fields, context=context)
-        if context.get('active_model') == 'res.partner' and context.get('active_ids'):
-            res.update({'partner_ids': context.get('active_ids')})
-        return res
+        self._compute_public_url()
+        self.public_url_html = '<a href="%s">%s</a>' % (self.public_url, _("Click here to start survey"))
```

Comment:
_compute_public_url_html is decorated with multi and assignation works on singleton